File: ueueo-framework/ueueo-background-jobs-abstractions/src/main/java/com/ueueo/backgroundjobs/AbpBackgroundJobOptions.cs
using System;
using System.Collections.Generic;
using System.Collections.Immutable;

namespace Volo.Abp.BackgroundJobs;

public class AbpBackgroundJobOptions
{
    private readonly Dictionary<Type, BackgroundJobConfiguration> _jobConfigurationsByArgsType;
    private readonly Dictionary<String, BackgroundJobConfiguration> _jobConfigurationsByName;

    /**
     * Default: true.
    */
    public boolean IsJobExecutionEnabled;// { get; set; } = true;

    public AbpBackgroundJobOptions()
    {
        _jobConfigurationsByArgsType = new Dictionary<Type, BackgroundJobConfiguration>();
        _jobConfigurationsByName = new Dictionary<String, BackgroundJobConfiguration>();
    }

    public BackgroundJobConfiguration GetJob<TArgs>()
    {
        return GetJob(typeof(TArgs));
    }

    public BackgroundJobConfiguration GetJob(Type argsType)
    {
        var jobConfiguration = _jobConfigurationsByArgsType.GetOrDefault(argsType);

        if (jobConfiguration == null)
        {
            throw new AbpException("Undefined background job for the job args type: " + argsType.AssemblyQualifiedName);
        }

        return jobConfiguration;
    }

    public BackgroundJobConfiguration GetJob(String name)
    {
        var jobConfiguration = _jobConfigurationsByName.GetOrDefault(name);

        if (jobConfiguration == null)
        {
            throw new AbpException("Undefined background job for the job name: " + name);
        }

        return jobConfiguration;
    }

    public IReadOnlyList<BackgroundJobConfiguration> GetJobs()
    {
        return _jobConfigurationsByArgsType.Values.ToImmutableList();
    }

    public void AddJob<TJob>()
    {
        AddJob(typeof(TJob));
    }

    public void AddJob(Type jobType)
    {
        AddJob(new BackgroundJobConfiguration(jobType));
    }

    public void AddJob(BackgroundJobConfiguration jobConfiguration)
    {
        _jobConfigurationsByArgsType[jobConfiguration.ArgsType] = jobConfiguration;
        _jobConfigurationsByName[jobConfiguration.JobName] = jobConfiguration;
    }
}

File: ueueo-framework/ueueo-background-jobs-abstractions/src/main/java/com/ueueo/backgroundjobs/AbpBackgroundJobOptions.java
package com.ueueo.backgroundjobs;

import com.ueueo.exception.BaseException;

import java.util.HashMap;
import java.util.Map;

public class AbpBackgroundJobOptions {
//    private Map<Class<?>, BackgroundJobConfiguration> jobConfigurationsByArgsType;
    private Map<String, BackgroundJobConfiguration> jobConfigurationsByName;

    /**
     * Default: true.
     */
    private boolean isJobExecutionEnabled = true;

    public AbpBackgroundJobOptions() {
//        jobConfigurationsByArgsType = new HashMap<>();
        jobConfigurationsByName = new HashMap<>();
    }

//    public BackgroundJobConfiguration getJob(Class<?> argsType) {
//        BackgroundJobConfiguration jobConfiguration = jobConfigurationsByArgsType.get(argsType);
//
//        if (jobConfiguration == null) {
//            throw new AbpException("Undefined background job for the job args type: " + argsType.getName());
//        }
//
//        return jobConfiguration;
//    }

    public BackgroundJobConfiguration getJob(String name) {
        BackgroundJobConfiguration jobConfiguration = jobConfigurationsByName.get(name);

        if (jobConfiguration == null) {
            throw new BaseException("Undefined background job for the job name: " + name);
        }

        return jobConfiguration;
    }

//    public List<BackgroundJobConfiguration> getJobs() {
//        return new ArrayList<>(jobConfigurationsByArgsType.values());
//    }

    public void addJob(Class<?> jobType) {
        addJob(new BackgroundJobConfiguration(jobType));
    }

    public void addJob(BackgroundJobConfiguration jobConfiguration) {
//        jobConfigurationsByArgsType.put(jobConfiguration.getArgsType(), jobConfiguration);
        jobConfigurationsByName.put(jobConfiguration.getJobName(), jobConfiguration);
    }

    public boolean isJobExecutionEnabled() {
        return isJobExecutionEnabled;
    }

    public void setJobExecutionEnabled(boolean jobExecutionEnabled) {
        isJobExecutionEnabled = jobExecutionEnabled;
    }
}
